docs: Add k3d platform setup (#12642)
istio/istio.io: content/en/docs/setup/platform-setup/k3d/index.md

---
title: k3d

> description: Instructions to set up k3d for Istio.

Author: @ericvn  fixed 0a8d553bf999b6acc59b0bf64b152f6546b00ff0

> weight: 30

Reviewer: The weight needs to place the page alphabetically and should no be the same as another page. the `kind` page uses 30 already.

Author: @ericvn fixed 0e2a14c3ed1e2e18084ca60f32ba9f190e3df456

skip_seealso: true
keywords: [platform-setup,kubernetes,k3d,k3s]
owner: istio/wg-environments-maintainers
test: no
---

> K3D is a lightweight wrapper to run [K3S](https://github.com/rancher/k3s) (Rancher Lab’s minimal Kubernetes distribution) in docker.

Author: @ericvn This description is taken from the official website  https://k3d.io/v5.4.7/#what-is-k3d

k3d makes it very easy to create single- and multi-node k3s clusters in docker, e.g. for local development on Kubernetes.

## Prerequisites

- To use k3d, you will also need to [install docker](https://docs.docker.com/install/).
- Install the latest version of [k3d](https://k3d.io/v5.4.7/#installation).
- To interact with the Kubernetes cluster [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl)
- (Optional) [Helm](https://helm.sh/docs/intro/install/) is the package manager for Kubernetes

> ## Installation steps

Reviewer 2: ```suggestion
## Installation
```

Author: @windsonsea fixed d12cc24fc105e41ea199fc305772d9e1fa204387

    {{< /text >}}

1.  To see the list of k3d clusters, use the following command:

    {{< text bash >}}
    $ k3d cluster list
    k3s-default
    {{< /text >}}

1.  To list the local Kubernetes contexts, use the following command.

    {{< text bash >}}
    $ kubectl config get-contexts
    CURRENT   NAME                 CLUSTER              AUTHINFO             NAMESPACE
    *         k3d-k3s-default      k3d-k3s-default      k3d-k3s-default
    {{< /text >}}

    {{< tip >}}
    `k3d-` is prefixed to the context and cluster names, for example: `k3d-k3s-default`
    {{< /tip >}}

1.  If you run multiple clusters, you need to choose which cluster `kubectl` talks to. You can set a default cluster
    for `kubectl` by setting the current context in the [Kubernetes kubeconfig](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/) file. Additionally you can run following command
    to set the current context for `kubectl`.

    {{< text bash >}}
    $ kubectl config use-context k3d-k3s-default
    Switched to context "k3d-k3s-default".
    {{< /text >}}

> ## Setup Istio for K3d

Reviewer 2: ```suggestion
## Set up Istio for K3d

```
Provide a blank line after the heading to avoid lint errors.

Author: @windsonsea fixed d12cc24fc105e41ea199fc305772d9e1fa204387

1.  Once you are done setting up a k3d cluster, you can proceed to [install Istio](docs/setup/install/helm/)
    on it.

    {{< text bash >}}
    $ kubectl create namespace istio-system
    $ helm install istio-base istio/base -n istio-system --wait
    $ helm install istiod istio/istiod -n istio-system --wait
    {{< /text >}}

1.  (Optional) Install an ingress gateway:

    {{< text bash >}}
    $ kubectl label namespace istio-system istio-injection=enabled
    $ helm install istio-ingressgateway istio/gateway -n istio-system --wait
    {{< /text >}}

> ## Unnstallation step

Reviewer 2: ```suggestion
## Uninstallation
```

Author: @windsonsea fixed d12cc24fc105e41ea199fc305772d9e1fa204387

1.  When you are done experimenting and you want to delete the existing cluster, use the following command:

    {{< text bash >}}
    $ k3d cluster delete k3s-default
    Deleting cluster "k3s-default" ...
    {{< /text >}}

> ## Setup Dashboard UI for K3d

Reviewer 2: ```suggestion
## Set up Dashboard UI for K3d
```

Author: @windsonsea fixed d12cc24fc105e41ea199fc305772d9e1fa204387

> k3d does not have a built in Dashboard UI like minikube. But you can still setup Dashboard, a web based Kubernetes UI, to view your cluster.

Reviewer 2: ```suggestion
k3d does not have a built in Dashboard UI like minikube. But you can still set up Dashboard, a web based Kubernetes UI, to view your cluster.
```

Author: @windsonsea fixed 130132885d005ac0b84a5191ddb11bcf98f8aa3d

    {{< /tip >}}

1.  If you run multiple clusters, you need to choose which cluster `kubectl` talks to. You can set a default cluster
    for `kubectl` by setting the current context in the [Kubernetes kubeconfig](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/) file. Additionally you can run following command
    to set the current context for `kubectl`.

    {{< text bash >}}
    $ kubectl config use-context k3d-k3s-default
    Switched to context "k3d-k3s-default".
    {{< /text >}}

## Set up Istio for K3d

1.  Once you are done setting up a k3d cluster, you can proceed to [install Istio](/docs/setup/install/helm/)
    on it.

    {{< text bash >}}
    $ kubectl create namespace istio-system
    $ helm install istio-base istio/base -n istio-system --wait
    $ helm install istiod istio/istiod -n istio-system --wait
    {{< /text >}}

1.  (Optional) Install an ingress gateway:

    {{< text bash >}}
    $ kubectl label namespace istio-system istio-injection=enabled
    $ helm install istio-ingressgateway istio/gateway -n istio-system --wait
    {{< /text >}}

> ## Uninstallation

Reviewer 3: move to the bottom of this page

Author: @zirain fixed 1bfeeeafe049f7350e8d81d40a0d241fc9d2bd42
